### ISSUE TITLE
Validate dynamic props in class

### ISSUE THREAD

**Issue by xvs32x** [typestack/class-validator#268]
Hi all, I have dynamic fields in form payload, for example phone, adrress and others . These fields can be added or deleted from form by the manager in the admin panel. And, I haven't any idea, how I can validate them. Can anyone share their experience in this matter? Big thanks!:)

**Comment by lgarczyn** [comment]
For the record, the solution is to add a group for each option, and use the groups you care about when validating, which can be done dynamically.

**Comment by xvs32x** [comment]
@lgarczyn Yes, You're right. But the basic thing I asked, how we can add props dynamically :) For example, when we have some form field in our database (and can change them from the interface (add, edit, remove and so on)). And we also have to validate them.

**Comment by lgarczyn** [comment]
The easiest way is to have all those components in a non-validated child object, that is parsed into a Map, and checked manually if it fits your database description of allowed fields.

But the best way to do it is with a custom validator. You can lookup the sources for most current validator to see how they work, though some directly affect the metadata, which you cannot do without access to internal classes. Basically use registerValidator.

Your validator can either pull data from the database using an async validator, or pull data from another property of the object being validated.

The dynamic values would need to stored in a child object, which will only be validated by your custom validator. It would simply iterate on all values of that object, and check that they follow your db declaration. You can call the pure validation functions of this library from that validator.

I'm not sure how nested validation occurs if you want to check classes inside these fields, but you would need to check how ValidateNested does it.

**Comment by NoNameProvided** [comment]
`class-validator` is not well suited to handle this. If your entity is dynamic and cannot be described before usage then I think currently this library is not a good fit for you. 

You can use a JSON schema-based validation library like  https://github.com/colinhacks/zod or https://github.com/hapijs/joi.
